.github/workflows/abstractions-python.yml

```diff
@@ -0,0 +1,73 @@
+name: Python abstractions
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+    paths: ['abstractions/python/**', '.github/workflows/**']
+  pull_request:
+    paths: ['abstractions/python/**', '.github/workflows/**']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      relativePath: ./abstractions/python
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.4.0
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pipenv
+          pipenv install --dev --skip-lock
+        working-directory: ${{ env.relativePath }}
+      - name: Check code format
+        run: |
+          pipenv run yapf -dr kiota/abstractions
+        working-directory: ${{ env.relativePath }}
+      - name: Check import order
+        run: |
+          pipenv run isort kiota/abstractions
+        working-directory: ${{ env.relativePath }}
+      - name: Lint with Pylint
+        run: |
+          pipenv run pylint kiota/abstractions --disable=W --rcfile=.pylintrc
+        working-directory: ${{ env.relativePath }}
+      - name: Static type checking with Mypy
+        run: |
+          pipenv run mypy kiota/abstractions
+        working-directory: ${{ env.relativePath }}
+
+  publish:
+    name: Publish distribution to PyPI
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    needs: [build]
+    env:
+      relativePath: ./abstractions/python
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install flit
+        run: |
+          pip install flit
+        working-directory: ${{ env.relativePath }}
+      - name: Publish the distibution to PyPI
+        run: flit publish
+        working-directory: ${{ env.relativePath }}
+        env:
+          FLIT_INDEX_URL: https://upload.pypi.org/legacy/
```

Comment:
is this the public feed?

Reply:
Yes, that is the link to public index

Reply:
I don't think we should publish to the public index just yet in case we need to make breaking changes. Github packages [doesn't seem to support Python](https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#support-for-package-registries).
May I suggest we create a new a ADO package feed and use that instead for the time being? https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_packaging?_a=feed&feed=PowerShellSDK_BuildFeed

Reply:
Ok. What if we publish to the test Index `https://test.pypi.org` for now then?

Reply:
this one is public I assume? but it's understood by the community that things on this index are not meant for production, correct? and can we pull a package entirely from that index? (I'm thinking of the case where we change the name of the package).
If the answer to both question is yes, please proceed.

Reply:
Thanks for making the change! Let me know if you need tme to add secrets to the repo

Reply:
I will. I currently don't have the privileges to add the secrets myself (can't access the project settings).